src/pe/debug.rs

```diff
@@ -131,60 +326,127 @@ pub struct CodeviewPDB70DebugInfo<'a> {
 }
 
 impl<'a> CodeviewPDB70DebugInfo<'a> {
-    pub fn parse(bytes: &'a [u8], idd: &ImageDebugDirectory) -> error::Result<Option<Self>> {
+    pub fn parse(
+        bytes: &'a [u8],
+        idd: ImageDebugDirectoryIterator<'_>,
+    ) -> error::Result<Option<Self>> {
         Self::parse_with_opts(bytes, idd, &options::ParseOptions::default())
     }
 
     pub fn parse_with_opts(
         bytes: &'a [u8],
-        idd: &ImageDebugDirectory,
+        idd: ImageDebugDirectoryIterator<'_>,
```

Comment:
breaking change; is this avoidable?

Reply:
Yes, it is avoidable. This PR was initially designed and assumed subject to breaking change, while I'm pretty sure we could do that w/o actually breaking it. I give it a shot.

Reply:
@m4b Alright, everything looks improved and there are no explicit alloc's left there. On the other hand, a breaking `pub` change in `debug::DebugData::parse_with_opts_and_fixup` for TE seems inevitable at this time. We actually _can_ let them alone, but that's out of our original purpose to support _multiple_ debug data directories.

Perhaps, there's a two ideas:
1. Let the TE breaking changes alone and release the rest as minor changes,
   Then, do the TE breaking changes separately in a 0.10 rollup.
2. Or, rollup everything, including the breaking TE change, in this PR as-is.

Which one do you prefer?